code/modules/mob/living/carbon/human/species_types/mothmen.dm

```diff
@@ -47,3 +55,125 @@
+	var/obj/item/organ/wingcheck = H.getorgan(/obj/item/organ/wings/moth)
+	if(!wingcheck) //This is to stop easy organ farms
+		to_chat(H, "<span class='warning'>You don't have any wings to regenerate!</span>")
+		return
+	if(!HAS_TRAIT(H, TRAIT_MOTH_BURNT))
+		to_chat(H, "<span class='warning'>Your wings are fine as they are!</span>")
+		return
+	if(H.nutrition < COCOON_NUTRITION_AMOUNT)
+		to_chat(H, "<span class='warning'>You are too hungry to weave a cocoon!</span>")
+		return
+	H.visible_message("<span class='notice'>[H] begins to hold still and concentrate on weaving a cocoon...</span>", \
+	"<span class='notice'>You begin to focus on weaving a cocoon... (This will take [COCOON_WEAVE_DELAY / 10] seconds and you must hold still.)</span>")
+	H.adjustStaminaLoss(20, 0) //this is here to deter people from spamming it if they get interrupted
+	if(do_after(H, COCOON_WEAVE_DELAY, FALSE, H))
+		if(!ismoth(H))
+			to_chat(H, "<span class='warning'>You have lost your mandibles and cannot weave anymore!.</span>")
+			return
+		if(H.incapacitated())
+			to_chat(H, "<span class='warning'>You cannot weave a cocoon in your current state.</span>")
+			return
+		H.visible_message("<span class='notice'>[H] finishes weaving a cocoon!</span>", "<span class='notice'>You finish weaving your cocoon.</span>")
+		var/obj/structure/moth/cocoon/C = new(get_turf(H))
+		H.forceMove(C)
+		H.Sleeping(20, 0)
+		C.preparing_to_emerge = TRUE
+		H.apply_status_effect(STATUS_EFFECT_COCOONED)
+		//owner.log_message("[key_name(owner)] has finished weaving a cocoon at [AREACOORD(owner)]")
+		addtimer(CALLBACK(src, .proc/emerge, C), COCOON_EMERGE_DELAY, TIMER_UNIQUE)
+	else
+		to_chat(H, "<span class='warning'>You need to hold still in order to weave a cocoon!</span>")
+
+//Removes moth from cocoon, restores burnt wings
+/datum/action/innate/cocoon/proc/emerge(obj/structure/moth/cocoon/C)
+	for(var/mob/living/carbon/human/H in C.contents)
+		if(!H.has_status_effect(STATUS_EFFECT_COCOONED))
+			return
+		//if(H.dna.features["moth_wings"] == "Burnt Off") //this check seems redundant as burned wings are a roundstart option and sending a signal to clear a non-existing mood event doesn't cause any issues
+		SEND_SIGNAL(H, COMSIG_CLEAR_MOOD_EVENT, "burnt_wings")
+		//if(!ismoth(H))//mutation toxin is a thing, so this is here to prevent accidentally creating a mishmash of moth and whatever else
+			//return Note - this makes the transformed user NEVER wake up
+		if(ismoth(H))
+			REMOVE_TRAIT(H, TRAIT_MOTH_BURNT, "fire")
+		H.dna.species.handle_mutant_bodyparts(H)
+		H.dna.species.handle_body(H)
+	C.preparing_to_emerge = FALSE
+	qdel(C)
+
+/datum/status_effect/cocooned/tick()
+	owner.SetSleeping(20, TRUE)
+	owner.adjustBruteLoss(-(COCOON_HEAL_AMOUNT / (COCOON_EMERGE_DELAY)), 0)
+	owner.adjustFireLoss(-(COCOON_HEAL_AMOUNT / (COCOON_EMERGE_DELAY)), 0)
+	owner.adjust_nutrition(-(COCOON_NUTRITION_AMOUNT / (COCOON_EMERGE_DELAY)), 0)
+
+
+/obj/structure/moth/cocoon
```

Comment:
/obj/structure/moth is not a thing and doesn't really make sense to exist so just name it moth_cocoon